GoogleForCreators/web-stories-wp
# Delete element when it's completely outside of Canvas

## Summary
Checks if an element is fully outside of a Page after being moved by the user and removes it if so.


## Relevant Technical Choices

- Same logic for single- and multi-selection
- Also makes the check when an element is rotated / resized since this can also result in the element being left completely out of canvas and not being accessible anymore.


## To-do

- [x] Confirm using the same logic when resizing/rotating




## User-facing changes

- When an element is left outside of canvas after dragging/resizing/rotating, it's deleted.


## Testing Instructions
I
1. Add a shape
2. Drag it outside of canvas to different sides of the Page
3. Verify that the shape is deleted if it's fully outside of canvas.
4. Verify that if the shape is partially on the canvas, it's not deleted.

II
1. Add multiple elements
2. Multi-select the elements
3. Drag or rotate or resize the elements so that one (or more) of the multi-selected elements are left fully outside of canvas
4. Verify that the elements outside of canvas were deleted.


---



Fixes #445
